persistent-template/Database/Persist/TH.hs

```diff
@@ -1689,18 +1694,22 @@ liftAndFixKeys entityMap EntityDef{..} =
     |]
 
 liftAndFixKey :: EntityMap -> FieldDef -> Q Exp
-liftAndFixKey entityMap (FieldDef a b c sqlTyp e f fieldRef mcomments) =
-    [|FieldDef a b c $(sqlTyp') e f fieldRef' mcomments|]
+liftAndFixKey entityMap (FieldDef a b c sqlTyp e f fieldRef mcomments casc) =
+    [|FieldDef a b c $(sqlTyp') e f fieldRef' mcomments casc|]
   where
     (fieldRef', sqlTyp') = fromMaybe (fieldRef, lift sqlTyp) $
       case fieldRef of
         ForeignRef refName _ft -> case M.lookup refName entityMap of
-          Nothing -> Nothing
+          Nothing -> checkCascade
           Just ent ->
             case fieldReference $ entityId ent of
               fr@(ForeignRef _Name ft) -> Just (fr, lift $ SqlTypeExp ft)
-              _ -> Nothing
-        _ -> Nothing
+              _ -> checkCascade
+        _ -> checkCascade
+    checkCascade = case casc of
+      FieldCascade Nothing Nothing -> Nothing
+      _ -> error $ "cascade field is not allown for field " <> show a
```

Comment:
```suggestion
      _ -> reportError $ "cascade field is not allowed for field " <> show a
```

`reportError` is a slightly nicer facility for reporting errors in the `Q` Template Haskell type. Would require refactoring this up a bit - probably:

```haskell
liftAndFixKey ... casc) = do
    (fieldRef', sqlTyp') <- fromMaybe (fieldRef, lift sqlTyp) <$> checkCascade
    [|FieldDef 
  where
    checkCascade casc =
        case fieldRef of
            ForeignRef refName _ft -> ...
                Nothing -> 
                    case casc of ...
```